src/services/actions/userActions.ts

```diff
@@ -32,3 +35,103 @@ export async function makeRegularUser(user: { userId: string }) {
     revalidatePath('/', 'layout')
   }
 }
+
+export async function updateUserProfile(
+  data: ProfileDataValues,
+) {
+  const validatedData = profileDataSchema.parse(data)
+   try {
+    const {id,picture, name, phoneNumber, stopCommunication,email,...socialData } = validatedData
+    return await prisma.$transaction(async (tx) => {
+      const currentUser = await tx.user.findUnique({
+        where: { id: id },
+        select: { dataCollected: true }
+      });
+
+      const updatedUser = await tx.user.update({
+        where: { id: id },
+        data: {
+          name,
+          phoneNumber,
+          stopCommunication,
+          picture,
+          email,
+          dataCollected: {
+            ...(currentUser?.dataCollected as Record<string, unknown>),
+            ...socialData
+          }
+        }
+      });
+      revalidatePath('/user/edit');
+      return { success: true, user: updatedUser };
+    });
+  } catch (error) {
+    console.error('Error updating profile:', error);
+    throw new Error('Failed to update profile');
+  }
+}
+export async function getUserProfile() {
+  try {
+    const userCurrent = await getUser()
+    if (!userCurrent?.id) return null
+    const user = await prisma.user.findUnique({
+      where: { id: userCurrent.id },
+      select: {
+        id: true,
+        name: true,
+        email: true,
+        picture: true,
+        phoneNumber: true,
+        stopCommunication: true,
+        dataCollected: true
+      }
+    })
+    if (!user) return null 
+    const dataCollected = user.dataCollected as UserDataCollected
+    const cleanUserData = {
+      id: user.id,
+      name: user.name ?? '',
+      email: user.email ?? '',
+      picture: user.picture ?? '',
+      phoneNumber: user.phoneNumber ?? '',
+      stopCommunication: user.stopCommunication ?? false,   
+      company: dataCollected?.company ?? '',
+      position: dataCollected?.position ?? '',
+      shortDescription: dataCollected?.shortDescription ?? '',
+      linkedin: dataCollected?.linkedin ?? '',
+      x: dataCollected?.x ?? '',
+      instagram: dataCollected?.instagram ?? '',
+      website: dataCollected?.website ?? ''
+    }
+
+
+    return profileDataSchema.parse(cleanUserData)
+  } catch (error) {
+    console.error('Error in getUserProfile:', error)
+    if (error instanceof z.ZodError) {
+      console.error('Validation error:', error.errors)
+    }
+    return null
+  }
+}
+
+export async function deleteUserAccount() {
+  try {
+    const userCurrent = await getUser()
+    if (!userCurrent?.id) return { success: false, error: 'User not found' }
+
+    // Eliminar todas las sesiones asociadas al usuario
+    await prisma.session.deleteMany({
+      where: { userId: userCurrent.id }
+    })
+
+    // Eliminar la cuenta del usuario
+    await prisma.user.delete({
```

Comment:
we do soft deletes. this means instead of using delete we simply set deletedAt:new Date()

this helps us avoid issues with DB integrity in the future.